src/Compilers/Core/Portable/Diagnostic/TriggerDiagnosticDescriptor.cs

```diff
@@ -18,7 +18,7 @@ public class TriggerDiagnosticDescriptor : DiagnosticDescriptor
         /// <param name="customTags">Optional custom tags for the diagnostic. See <see cref="WellKnownDiagnosticTags"/> for some well known tags.</param>
         public TriggerDiagnosticDescriptor(string id, params string[] customTags)
             : base(id, title: "", messageFormat: "", category: "", defaultSeverity: DiagnosticSeverity.Hidden, isEnabledByDefault: true,
-                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable))
+                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable).Append(WellKnownDiagnosticTags.Trigger))
```

Comment:
+1 to @shyamnamboodiripad's argument. I may be missing some larger purpose of this extensibility point, but why would I extend from DiagnosticDescriptor if it's just going to get reduced back to a DiagnosticDescriptor in the end anyway? 

TriggerDiagnosticDescriptor is just setting the diagnostic up a particular way which could be done with another constructor or Create method on DiagnosticDescriptor, and there's only one spot in the engine that checks the type, which could just as easily check the tag, right? We could then seal DiagnosticDescriptor and provide some other data transfer mechanism (either tags are sufficient as Shyam says, or introduce something else).

Seems like you can get rid of a dangerous and unnecessary extensibility point by switching one type check to a tag check, which is a win all around.